pkg/connector/backfill.go

```diff
@@ -19,11 +20,28 @@ import (
 
 // getTakeoutID blocks until the takeout ID is available.
 func (t *TelegramClient) getTakeoutID(ctx context.Context) (takeoutID int64, err error) {
-	log := zerolog.Ctx(ctx).With().Str("function", "getTakeoutID").Logger()
+	// Always stop the takeout timeout timer
+	if t.stopTakeoutTimer != nil {
+		t.stopTakeoutTimer.Stop()
+	}
+
+	// After we get the takeout ID, reset the timer.
+	defer func() {
+		t.stopTakeoutTimer = time.AfterFunc(max(time.Hour, time.Duration(t.main.Bridge.Config.Backfill.Queue.BatchDelay*2)), sync.OnceFunc(func() { t.stopTakeout(ctx) }))
```

Comment:
I meant that this should be done after checking the takeout id, because FetchMessages also resets the timer, but I guess it doesn't make much difference